### PR TITLE
feat: plumb through context changes

### PR DESCRIPTION
**Pull request by guseggert**

### CODE REVIEW

File: util.go [hunk 1]
```diff
@@ -133,7 +134,7 @@ func list(fs *Filestore, verify bool, key mh.Multihash) *ListRes {
 
 func listAll(fs *Filestore, verify bool) (func() *ListRes, error) {
 	q := dsq.Query{}
-	qr, err := fs.fm.ds.Query(q)
+	qr, err := fs.fm.ds.Query(context.Background(), q)
 	if err != nil {
 		return nil, err
 	}
```
Comment (petar) [inline util.go:140]: Either the lambda returned below or ListAll() should take a context. Probably ok to leave it like this for now.

**Review by petar — APPROVED** [review]

**Comment by guseggert** [comment]
This is wrong b/c I didn't realize that go-ipfs consumes go-filestore@v0.0.3 whereas this is building on the breaking changes introduced in v1.0.0, so I need to backport this.

**Comment by aschmahmann** [comment]
We still need to have a v1 version of this for use in places like lotus

**Review by aschmahmann — CHANGES_REQUESTED** [review]
This isn't quite ready, it should match #55

**Comment by marten-seemann** [comment]
This should be a v2, as the API was changed.

**Comment by aschmahmann** [comment]
@marten-seemann we're not bumping these to v2 as right now there is a good deal of complexity around the v0 and v1 (both in use) of this repo and they need to be swappable (i.e. I need to be able to drop in v1 as a v0 replacement).

This choice is independent of, but related to, the decision to not release go-datastore as a v2 despite the breaking change with contexts. So we're knowingly doing the wrong thing.

**Review by aschmahmann — APPROVED** [review]

**Pull request merged**